.pre-commit-config.yaml

```diff
@@ -0,0 +1,8 @@
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.5
```

Comment:
Does this need to match the installed version of ruff? If so, I think this needs to be in the poetry devDevdependencies so it can be found in the projects venv. Otherwise, if two repositories get out of sync, this might start causing problems between major versions. Especially since ruff is <1.0 still and is likely to break configs as well.

Reply:
I had this discussion with Michele yesterday, we worked out that pre-commit installs it's own version of ruff anyway behind the scenes so pinning the version here is required.

Reply:
At least that means the github action and the pre-commit hook should always match (although, actually, having comments pointing both ways to make it harder to forget to update both at the same time might be helpful). It'll just be the local version that acts weirdly if a repository uses a version with a breaking change to the config.

Reply:
Yes, that's exactly what we said yesterday: at least for now, we can rely on a local brew install of pre-commit and ruff, as long as the pre-commit hook and the CI have the same version. @MatthewCane had VSC issue with the version of ruff installed in the virtualenv so it might not be a viable option.